lib/mix/lib/mix/tasks/test.ex

```diff
@@ -595,20 +606,50 @@ defmodule Mix.Tasks.Test do
 
     # Prepare and extract all files to require and run
     test_paths = project[:test_paths] || default_test_paths()
-    test_pattern = project[:test_pattern] || "*_test.exs"
-    warn_test_pattern = project[:warn_test_pattern] || "*_test.ex"
+    test_pattern = project[:test_pattern] || "*.{ex,exs}"
+    test_load_pattern = project[:test_load_pattern] || ~r/.*_test\.exs$/
+
+    test_warn_pattern =
+      Keyword.get(project, :test_warn_pattern, ~r/^(?!.*_helper\.exs$)(?:.*_test\.ex|.*\.exs)$/)
+
+    test_warn_ignore_files = project[:test_warn_ignore_files] || []
+
+    # Warn about deprecated configurations
+    if project[:test_pattern] do
+      Mix.shell().info(
+        "warning: the `:test_pattern` configuration is deprecated and will be ignored. Use `:test_load_pattern` instead."
+      )
+    end
+
+    if project[:warn_test_pattern] do
+      Mix.shell().info(
+        "warning: the `:warn_test_pattern` configuration is deprecated and will be ignored. Use `:test_warn_pattern` instead."
+      )
+    end
 
     {test_files, test_opts} =
       if files != [], do: ExUnit.Filters.parse_paths(files), else: {test_paths, []}
 
-    unfiltered_test_files = Mix.Utils.extract_files(test_files, test_pattern)
+    # get a list of all files in the test folders, which we filter by the test_load_pattern
+    potential_test_files = Mix.Utils.extract_files(test_files, test_pattern)
+
+    unfiltered_test_files =
+      Enum.filter(potential_test_files, &Regex.match?(test_load_pattern, &1))
```

Comment:
Do we need to filter them? Can't we just assume that everything we don't warn is going to be used?